units/systemd-resolved.service.in

```diff
@@ -16,9 +16,11 @@ Documentation=https://www.freedesktop.org/wiki/Software/systemd/writing-resolver
 
 DefaultDependencies=no
 After=systemd-sysusers.service systemd-networkd.service
+After=systemd-resolv-conf.service
 Before=network.target nss-lookup.target shutdown.target
 Conflicts=shutdown.target
 Wants=nss-lookup.target
+Wants=systemd-resolv-conf.service
```

Comment:
Don't make the dependency here unconditional. Add `[Install] Also=systemd-resolv-conf.service` below, and in the new service file, add `[Install] WantedBy=systemd-resolved.service`.